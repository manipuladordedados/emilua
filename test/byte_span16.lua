local byte_span = require 'byte_span'

local bs = byte_span.new(1, 3)
print(#bs)
print(bs.capacity)

bs:slice(1, 3):copy('foo')
print(bs)
print(bs:slice(1, 3))

print(bs:append('bar'))
print(bs:slice(1, 3))
print(bs:append('ba'))
print(bs:slice(1, 3))

bs = bs:append('oo')
print(bs:append(' ', bs, ' ', bs))

print(byte_span.append(nil, bs))
