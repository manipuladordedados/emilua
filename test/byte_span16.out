1
3
f
foo
fbar
foo
fba
fba
foo foo foo
